### PR TITLE
fix(util): handle nil values in AssertType

### PR DESCRIPTION
**Pull request by pushkarscripts**
## Description

Improve `AssertType` nil safety and type mismatch diagnostics in `tool/util/assert.go`.

This updates type formatting to use `reflect.Type.String()` and prevents nil-related panics during failed assertions. Also adds minimal unit coverage for concrete and pointer type assertions.

## Motivation

`AssertType` currently calls `reflect.TypeOf(v).Name()` directly, which can panic for nil values and produces unclear diagnostics for composite types like pointers.

This change makes assertion failures safer and improves error messages for debugging type mismatches.

---

## Checklist

- [x] PR title follows [conventional commits](https://www.conventionalcommits.org/) format
- [x] Code formatted: `make format`
- [x] Linters pass: `make lint`
- [x] Tests pass: `make test`
- [x] Tests added for new functionality
- [x] Tests follow [testing guidelines](docs/testing.md)
- [ ] Documentation updated (if applicable)

### CODE REVIEW

File: tool/util/assert_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAssertType_Success(t *testing.T) {
```
Comment (txabman42) [inline tool/util/assert_test.go:16]: Lets also test what this PR tries to fix: _handle nil values in AssertType_.

**Review by txabman42 — CHANGES_REQUESTED** [review]

File: tool/util/assert_test.go (hunk 1 above)
Comment (pushkarscripts) [reply tool/util/assert_test.go:16]: > Lets also test what this PR tries to fix: _handle nil values in AssertType_.

Good point, I’ll add coverage for nil handling as well

File: tool/util/assert.go [hunk 2]
```diff
@@ -18,10 +18,12 @@ func Assert(condition bool, message string) {
 func AssertType[T any](v any) T {
 	value, ok := v.(T)
 	if !ok {
-		actualType := reflect.TypeOf(v).Name()
-		var zero T
-		expectType := reflect.TypeOf(zero).String()
-		ex.Fatalf("Type assertion failed: %s, expected %s",
+		actualType := "nil"
```
Comment (kakkoyun) [inline tool/util/assert.go:21]: suggestion: We also have a new `TypeFor` API, it worth checking https://pkg.go.dev/reflect#TypeFor (I haven't check how it behaves with pointer types and nil).

nit: We can also strictly check if the type is a pointer type and then assume it could be nil.

Comment (pushkarscripts) [reply tool/util/assert.go:21]: > suggestion: We also have a new `TypeFor` API, it worth checking https://pkg.go.dev/reflect#TypeFor (I haven't check how it behaves with pointer types and nil).
> 
> nit: We can also strictly check if the type is a pointer type and then assume it could be nil.

sure, I’ll update it accordingly.

**Review by kakkoyun — APPROVED** [review]

**Review by txabman42 — APPROVED** [review]
LGTM! TY ❤️
